src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResource.kt

```diff
@@ -185,6 +185,67 @@ class ActivitiesResource(private val activitiesService: ActivitiesService) {
   fun createOffenderProgramProfile(
     @Schema(description = "Course activity id", required = true) @PathVariable courseActivityId: Long,
     @RequestBody @Valid createRequest: CreateOffenderProgramProfileRequest
-  ): CreateOffenderProgramProfileResponse =
+  ): OffenderProgramProfileResponse =
     activitiesService.createOffenderProgramProfile(courseActivityId, createRequest)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_ACTIVITIES')")
+  @PutMapping("/activities/{courseActivityId}/booking-id/{bookingId}/end")
```

Comment:
i called it end as it basically just sets an end date with an optional reason and comment

Reply:
As ever with REST we could implement many ways, but It feels like a delete to me - we're removing a booking from a course.